fbw-a32nx/src/systems/fmgc/src/flightplanning/plans/BaseFlightPlan.ts

```diff
@@ -2109,31 +2110,40 @@ export abstract class BaseFlightPlan<P extends FlightPlanPerformanceData = Fligh
         const xiToXf = lastLegInFirst.isXI() && element.isXF();
 
         if (xiToXf) {
-          if (LnavConfig.VERBOSE_FPM_LOG) {
-            console.log(`[fpm] stringSegmentsForwards - cutBefore (xiToXf)) = ${i}`);
-          }
+          // If the last leg in the first segment is a PI leg, we need to check if the waypoint is the same as the one in the second segment
+          if (
+            lastLegInFirst.type !== LegType.PI ||
+            areDatabaseItemsEqual(lastLegInFirst.definition.waypoint, element.definition.waypoint)
+          ) {
+            if (LnavConfig.VERBOSE_FPM_LOG) {
+              console.log(`[fpm] stringSegmentsForwards - cutBefore (xiToXf)) = ${i}`);
+            }
 
-          // Convert TF to CF
-          if (element.type === LegType.TF) {
-            const prevElement = second.allLegs[i - 1];
-            if (!prevElement || prevElement.isDiscontinuity === true) {
-              throw new Error('[FMS/FPM] TF leg without a preceding leg');
-            } else if (!prevElement.terminationWaypoint()) {
-              throw new Error('[FMS/FPM] TF leg without a preceding leg with a termination waypoint');
+            // Convert TF to CF
+            if (element.type === LegType.TF) {
```

Comment:
If the leg before is a PI this should *never* be hit, as PI-CF must always and can only ever exist as a combo from an approach transition.

Reply:
Yes, this `if` applies when the leg before is not a PI, for example, when the approach transition ends in a CI

Reply:
It applies if there's a PI, then a CF with the same fix (the or condition)?